Source/Core/Core/HW/WiimoteEmu/Speaker.cpp

```diff
@@ -106,18 +106,17 @@ void Wiimote::SpeakerData(wm_speaker_data* sd)
 		return;
 	}
 
-	// Speaker Pan
-	unsigned int vol = (unsigned int)(m_options->settings[4]->GetValue() * 100);
-
+	// Speaker Pan using Constant Power Pan Law
+	double pan = m_options->settings[4]->GetValue() / 1.27;
+	double pan_prime = 3.14159265358979323846 * (pan + 1) / 4.0;
```

Comment:
We should probably have our own separate constants header that type-specializes in a generic way like numeric_limits.

e.g.

```cpp
template <typename T>
class Constants
{
};

template <>
class Constants<float>
{
public:
	static constexpr float Pi()
	{
		return /* floating precision here */;
	}
};

template <>
class Constants<double>
{
public:
	static constexpr double Pi()
	{
		return /* double precision here */;
	}
};
```

which can then be called as

```cpp
Constants<double>::Pi()
or
Constants<float>::Pi()
```